queries.sql

```diff
@@ -8,3 +8,65 @@ SELECT name,escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = TRUE;
 SELECT * FROM animals WHERE name NOT IN ('Gabumon');
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+/*Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.*/
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+/*Inside a transaction:
+Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+Verify that changes were made.
+Commit the transaction.
+Verify that changes persist after commit.*/
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE ('%mon');
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+
+/*Inside a transaction delete all records in the animals table, then roll back the transaction.
+After the rollback verify if all records in the animals table still exists.)*/
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```

Comment:
- [ Required ] I think it would be nice that you write a `SELECT` query before doing the `Rollback` just to confirm that the `DELETE` query was successful. Please see the example below as a suggestion of how you can acieve this.

## Example code
```SQL
   BEGIN;

   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```